File: .github/fixtures/profiles/mcp-enabled/src/text_stats.rb
# frozen_string_literal: true

module TextStatsMcp
  VERSION = "1.0.0"
  PROTOCOL_VERSION = "2025-11-25"
  TOOL_NAME = "text_stats"

  module_function

  def analyze(text)
    {
      "bytes" => text.bytesize,
      "lines" => text.empty? ? 0 : text.lines.count,
      "words" => text.scan(/\S+/).length
    }
  end
end

File: .github/fixtures/profiles/mcp-enabled/src/text_stats.mjs
export function textStats(text) {
  if (typeof text !== "string") {
    throw new TypeError("text must be a string");
  }

  const bytes = Buffer.byteLength(text, "utf8");
  const lines = text.length === 0 ? 0 : text.split(/\n/).length;
  const trimmed = text.trim();
  const words = trimmed === "" ? 0 : trimmed.split(/\s+/u).length;

  return { bytes, lines, words };
}
